src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue

```diff
@@ -698,8 +698,12 @@ export default {
 		// const basketItems = this.rawPageData?.shop?.basket?.items?.values ?? [];
 		this.loadingPage = basketItems.some(item => item.__typename === 'LoanReservation'); // eslint-disable-line no-underscore-dangle, max-len
 	},
-	mounted() {
+	async mounted() {
 		this.$root.$on('jumpToLoans', this.jumpToLoans);
+
+		// check for applied promo
+		await this.verifyOrApplyPromotion();
```

Comment:
This method starts our data aggregation for the promo state, we want to control this flow deliberately so we have all the promo information before proceeding with other steps.